x-pack/plugins/apm/public/hooks/use_environments_fetcher.tsx

```diff
@@ -17,10 +17,16 @@ function getEnvironmentOptions(environments: string[]) {
     .filter((env) => env !== ENVIRONMENT_NOT_DEFINED.value)
     .map((environment) => ({
       value: environment,
-      text: environment,
+      label: environment,
     }));
 
-  return [ENVIRONMENT_ALL, ...environmentOptions];
+  return [
+    ENVIRONMENT_ALL,
+    ...(environments.includes(ENVIRONMENT_NOT_DEFINED.value)
+      ? [ENVIRONMENT_NOT_DEFINED]
+      : []),
```

Comment:
does this mean we lose the sort? is the list sorted usually, and is it important? (i'd say it's okay to not care much for sorting here, as long as we've considered the implications).

Reply:
The environment list will be displayed as it used to be, the sorting remains unchanged. 

We didn't specify a sort option for the [endpoint](https://github.com/elastic/kibana/blob/main/x-pack/plugins/apm/server/routes/environments/get_environments.ts#L23) so I think we can discuss the sorting in another ticket